RFCs/0004-pluggable-monitor.md

```diff
@@ -0,0 +1,453 @@
+monitor.required.PROTOCOL2=PLATFORM:MONITOR_NAME2
+...
+```
+
+in our specific example the directive should be:
+
+```
+monitor.required.ble=arduino:ble-monitor
+```
+
+where `ble` is the port protocol identification returned by the matching pluggable discovery.
+
+#### Using a monitor tool made by a 3rd party
+
+A platform developer may opt to depend on a monitor tool developed by a 3rd party instead of writing and maintaining his own.
+
+Since writing a good-quality cross-platform monitor tool is very hard and time consuming, we expect this option to be the one used by the majority of the developers.
+
+#### Direct monitor tool integration (not recommended)
+
+A monitor tool may be directly added to a platform, without passing through the `discoveryDependencies` in `package_index.json`, using the following directive in the `platform.txt`:
+
+```
+monitor.pattern.PROTOCOL=MONITOR_RECIPE
+```
+
+where `MONITOR_RECIPE` must be replaced by the command line to launch the monitor tool for the specific `PROTOCOL`. An example could be:
+
+```
+monitor.pattern.custom-ble="{runtime.tools.my-ble-monitor.path}/my-ble-monitor" -H
+```
+
+in this case the platform provides a new `custom-ble` protocol monitor tool and the command line tool named `my-ble-monitor` is launched with the `-H` parameter to start the monitor tool. In this case the command line pattern may contain any extra parameter in the formula: this is different from the monitor tools installed through the `discoveryDependencies` field that must run without any command line parameter.
+
+This kind of integration may turn out useful:
+
+- during the development of a platform (because providing a full `package_index.json` may be cumbersome)
+- if the monitor tool is specific for a platform and can not be used by 3rd party
+
+Anyway, since this kind of integration does not allow reusing a monitor tool between different platforms, we do not recommend its use.
+
+#### built-in monitor tools and backward compatibliity considerations
+
+Some monitor tools like the Arduino `serial-monitor` or the Arduino `network-monitor` must be always available, so they will be part of the `builtin` package and installed without the need to be part of a real package (`builtin` is a dummy package that we use to install tools that are not part of any platforms like `ctags` for example).
+
+If a platform requires the builtin monitor tools it must declare it with:
+
+```
+monitor.required.serial=builtin:serial-monitor
+monitor.required.network=builtin:network-monitor
+```
+
+For backward compatibility, if a platform does not declare any discovery or monitor tool (using the `discovery.*` or `monitor.*` properties in `platform.txt` respectively) it will automatically inherits `builtin:serial-monitor` and `builtin:network-monitor` (but not other `builtin` monitor tools that may be possibly added in the future). This will allow all legacy non-pluggable platforms to migrate to pluggable monitor without disruption.
```

Comment:
```suggestion
For backward compatibility, if a platform does not declare any discovery or monitor tool (using the `discovery.*` or `monitor.*` properties in `platform.txt` respectively) it will automatically inherit `builtin:serial-monitor` and `builtin:network-monitor` (but not other `builtin` monitor tools that may be possibly added in the future). This will allow all legacy non-pluggable platforms to migrate to pluggable monitor without disruption.
```